src/tree-view/treeViewComponent.ts

```diff
@@ -0,0 +1,81 @@
+const $ = (<any> window).$;
+const Promise = (<any> window).Promise;
+
+export class TreeViewController {
+  public tree;
+  public reselect;
+  public data;
+  public name : string;
+  public onSelect;
+  public lazyLoad;
+
+  /*@ngInject*/
+  constructor(private $element, private $timeout, private $http) {}
+
+  public $onInit() {
+    let element = this.$element[0].querySelector('div.treeview');
+    this.renderTree(element).then(() => {
+      this.tree = $(element).treeview(true);
+
+      this.tree.getNodes().forEach((node) => {
+        if (this.getTreeState(node) === !node.state.expanded) {
+          this.tree.toggleNodeExpanded(node);
+        }
+      });
+    });
+  }
+
+  private renderTree(element) {
+    return new Promise((resolve) => {
+      $(element).treeview({
+        data:            this.data,
+        expandIcon:      'fa fa-fw fa-angle-right',
+        collapseIcon:    'fa fa-fw fa-angle-down',
+        loadingIcon:     'fa fa-fw fa-spinner fa-pulse',
+        levels:          1,
+        allowReselect:   this.reselect,
+        showBorders:     false,
+        onNodeExpanded:  this.setTreeState(true),
+        onNodeCollapsed: this.setTreeState(false),
+        onNodeSelected:  (_event, node) => this.onSelect(node),
+        lazyLoad:        (node, render) => this.lazyLoad(node).then(render),
```

Comment:
Even tho it's nice to read promises, can we expose public API with promises? I am okay with local calling of promises, but public API shouldn't expose such thing. Consider changing `lazyLoad` to be [callback_binding](http://www.codelord.net/2016/05/13/understanding-angulars-and-binding/). Same applies to `onSelect` it's not a object, but `callback_binding`.